R/verb-joins.R

```diff
@@ -435,6 +464,43 @@ join_vars <- function(x_names, y_names, type, by, suffix = c(".x", ".y"), call =
   )
 }
 
+join_simple_table_alias <- function(table_names, aliases) {
+  stopifnot(length(table_names) == 2)
+  stopifnot(length(aliases) == 2)
+  aliases <- unlist(aliases)
+
+  x_alias <- aliases[[1]]
+  y_alias <- aliases[[2]]
+  x_name <- table_names[[1]]
+  y_name <- table_names[[2]]
+
+  x_out <- dplyr::coalesce(x_alias, x_name, "LHS")
```

Comment:
I think of `coalesce()` as a vector operation, so this feels a bit off to me. I'd prefer to see `x <- x_out %||% x_alias %||% x_name %||% "LHS"` but I can see why that would be painful to implement given the current structure. I'm not sure whether or not that suggests some bigger changes are necessary.

Reply:
This was actually quite simple to change. I used `coalesce()` due to its origin in the multi join PR. But I also prefer using `%||%`. Let's see whether this will be annoying for the multi join PR or not.